components/sso-saml/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/validators/SPInitSSOAuthnRequestValidator.java

```diff
@@ -155,7 +155,10 @@ public SAMLSSOReqValidationResponseDTO validate() throws IdentityException {
             validationResponse.setValid(true);
             validationResponse.setPassive(authnReq.isPassive());
             validationResponse.setForceAuthn(authnReq.isForceAuthn());
-
+            Integer index = authnReq.getAttributeConsumingServiceIndex();
```

Comment:
authnReq is an opensaml class that returns Integer type. Since it can return null value which is not auto unboxed had to keep it as an Integer here. In the below lines it will be auto unboxed and treated as an int.